src/async_impl/client.rs

```diff
@@ -169,6 +171,14 @@ impl ClientBuilder {
                 headers.get(USER_AGENT).cloned()
             }
 
+            let http = match config.trust_dns {
+                false => HttpConnector::new_gai(),
+                #[cfg(feature = "trust-dns")]
+                true => HttpConnector::new_trust_dns()?,
+                #[cfg(not(feature = "trust-dns"))]
+                true => unreachable!("trust-dns shouldn't be enabled unless the feature is"),
+            };
```

Comment:
Maybe there is some cleaner way to do this?

Reply:
Looks fine to me!